### ISSUE TITLE
Useless super delegation in method when overriding abstract methods.

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#1594]
```
from contextlib import AbstractContextManager

class ProgramContext(AbstractContextManager):

    # Overridden methods ------------------------------------------------------
    # Overrides method from AbstractContextManager.
    def __exit__(self, exc_type, exc_value, traceback):
        super().__exit__(exc_type, exc_value, traceback)
```

This gives "useless super delegation", but this method implements an abstract method from from AbstractContextManager.  The super delegation is required if this class is to participate in a polymorphic hierarchy since it doesn't know where in the inheritance chain it will be inherited.

**Comment by PCManticore** [comment]
I'm not sure this makes sense. Yes, the original method is abstract and needs to be implemented, but in this case the implementation is moot, since in doesn't do anything but delegating to an empty implementation of in the base class. It actually works because the original ``__exit__`` method doesn't do anything and calling an abstract method doesn't raise any TypeError whatsoever.

**Comment by NeilGirdhar** [comment]
It's not moot.  Without this definition, the class cannot be instantiated.

**Comment by NeilGirdhar** [comment]
(Please reopen this bug in pylint.)

**Comment by PCManticore** [comment]
Please disable the warning locally, this is not a bug in pylint.

**Comment by NeilGirdhar** [comment]
It is a bug.  The error should be suppressed when the method being overridden is marked abstract.  You can test for this using `if getattr(method, "__isabstractmethod__", False)}`.

**Comment by NeilGirdhar** [comment]
@PCManticore Hi, this bug is still in master.  Can you please try instantiating a class that derives from `AbstractContextManager` to verify that it cannot be done without this simple override?  This error should not be reported.
